server/server.js

```diff
@@ -30,6 +30,14 @@ app.post('/purchase', (req, res) => {
   //STRETCH FEATURE: authentication for customer
 })
 
+app.use(function (req, res, next) {
+  //404
+  res.locals.message = 'PAGE NOT FOUND';
+  const err = new Error('RESOURCE NOT FOUND');
+  err.status = 404;
+  return next(err);
```

Comment:
Here you're invoking an error handler but you have not provided an error handler yet